### DISCUSSION TITLE
[BUG] - Mealie does not start on synology after latest update

### DISCUSSION THREAD

**Discussion by HiiiiiHa** [mealie-recipes/mealie#3081] [General]
### First Check

- [X] This is not a feature request.
- [X] I added a very descriptive title to this issue (title field is above this).
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [ ] This issue can be replicated on the demo site (https://demo.mealie.io/).

### What is the issue you are experiencing?

I'm using mealie docker container on my Synology. My NAS is recent and DSM up-to-date.
I recently updated the Mealie image and Mealie stopped working
Failure occurs each time I try to start the container

### Steps to Reproduce

1) update mealie container to latest version
2) container no more starts

### Please provide relevant logs

/bin/sh: 1: /app/mealie/run.sh: not found

### Mealie Version

latest from https://registry.hub.docker.com/r/hkotel/mealie/

### Deployment

Docker (Synology)

**Comment by boc-the-git** [comment]
Provide your docker-compose file please

**Comment by HiiiiiHa** [comment]
I'm using the Container manager hence do not have a docker-compose file, anyway here is the export of the container configuration
[mealie.json](https://github.com/mealie-recipes/mealie/files/14096301/mealie.json)

**Comment by boc-the-git** [comment]
OK, I don't see anything in there of too much interest to me.

Ultimately, the error message is valid: `/app/mealie/run.sh: not found` makes perfect sense, because the file would be located at `/app/run.sh`, via lines 22 and 138 [here](https://github.com/mealie-recipes/mealie/blob/mealie-next/docker/Dockerfile).

The question is why is your system trying to access them at the other path, and I can't give a definitive answer there. I would've been in a better position to help if you were using docker-compose.

Some of our Synology users might be able to assist though.
For the record, I'm certain we have people using Synology without issue, so this isn't an issue impacting all Synology users.

**Comment by Kuchenpirat** [comment]
you propably upgraded from version 0.5.6. 
just in case make at least a backup of your data. 

Try chaning the image to v0.5.6. If it boots use the migration guide here: https://nightly.mealie.io/documentation/getting-started/migrating-to-mealie-v1/ to upgrade to the newest version

**Comment by HiiiiiHa** [comment]
Thanks for having a look. What is surprising is that issue appeared only with last update(I'm updating frequently). If I'm the only one I will reconfigure from scratch

**Comment by Kuchenpirat** [comment]
what version do you have mealie pinned to? latest, nightly or a specific version?
the latest tag would have had a huge bump from 0.5.6 (about two years old) to the v1 release

**Comment by boc-the-git** [comment]
> Thanks for having a look. What is surprising is that issue appeared only with last update(I'm updating frequently). If I'm the only one I will reconfigure from scratch

Depending what tag you're using/updating to, there might've been no updates. If you're using the `latest` tag for instance, it went almost 2 years without an update until ~2 weeks ago.

**Comment by davideb263** [comment]
I have a similar problem too on Synology using container manager and watchtower for auto updates. 
Watchtower updated to the latest image on Jan-28 and after that the container starts but the web interface is not accessible anymore.

**Comment by Kuchenpirat** [comment]
> I have a similar problem too on Synology using container manager and watchtower for auto updates. Watchtower updated to the latest image on Jan-28 and after that the container starts but the web interface is not accessible anymore.

pinn your version to v0.5.6 restart the container and follow the migration guide. New synologies support using docker compose directly and you could use this docker compose file: https://nightly.mealie.io/documentation/getting-started/installation/sqlite/

Otherwise you would have to propagate the values in the synology container manager based on this compose file

**Comment by HiiiiiHa** [comment]
after deleting image and container, and recreating from scratch:
- Mealie is now running
- I cannot login anymore nor request password reset nor login with default user\pwd :(

**Comment by davideb263** [comment]
> > I have a similar problem too on Synology using container manager and watchtower for auto updates. Watchtower updated to the latest image on Jan-28 and after that the container starts but the web interface is not accessible anymore.
> 
> pinn your version to v0.5.6 restart the container and follow the migration guide. New synologies support using docker compose directly and you could use this docker compose file: https://nightly.mealie.io/documentation/getting-started/installation/sqlite/
> 
> Otherwise you would have to propagate the values in the synology container manager based on this compose file

I rollbacked to v0.5.6 version and now is running again. I will migrate to v1.0 soon. Thanks for the help :)

**Comment by Kuchenpirat** [comment]
> after deleting image and container, and recreating from scratch:
> 
> * Mealie is now running
> * I cannot login anymore nor request password reset nor login with default user\pwd :(

is there anything in the logs?

**Comment by HiiiiiHa** [comment]
here it is
![1](https://github.com/mealie-recipes/mealie/assets/81105552/3c1427a8-31e3-44a3-8111-caa4c78dfa05)

**Comment by Kuchenpirat** [comment]
ok, did you by recreating also kill your data?
I have no idea how synologies work, so im just guessing at this point?
But it seems as there are things missing in you db that should be there

**Comment by HiiiiiHa** [comment]
I did not touched my local data. Only deleted container and image. Then recreated pointing to my existing data.

**Comment by Kuchenpirat** [comment]
> Thanks for having a look. What is surprising is that issue appeared only with last update(I'm updating frequently). If I'm the only one I will reconfigure from scratch

just to make sure, which tag have you used before? From the json you send i guess it is an earlier version because you are using port 80, which we have not done since a long time, but none the less missing db tables should not just appear

**Comment by michael-genson** [comment]
Moving this to a discussion since this is deployment-specific

@HiiiiiHa - it sounds like you upgraded from 0.5.6 to 1.0.0 by pinning the latest tag. Simply upgrading your instance won't work, you will need to:
1. Spin up an old version of Mealie (select tag 0.5.6, instead of latest)
2. Get a backup from your old version (https://nightly.mealie.io/documentation/getting-started/migrating-to-mealie-v1/)
3. Completely destroy your old version, including both the container and the data (grab a backup beforehand) OR follow the next steps in a different directory (do not try to use the existing directory unless you remove all of the data)
4. Spin up a fresh new instance of v1 (you can use a docker compose file from the docs and use the Projects tab on the docker app: https://nightly.mealie.io/documentation/getting-started/installation/installation-checklist/)
5. Follow the procedure for migration (see my first link)

**Comment by HiiiiiHa** [comment]
Thanks this worked for me. I had 2 instances of Mealie running at same time(one v0.5.6 and one v1), exported and imported in V1
Now it is working fine again
thanks

**Comment by eniac2000** [comment]
Hi everybody,
I have the same problem. I did the update in the container manager of DSM to the newest version of Mealie but it does not work.
How can I roll back to do a backup? Afterwards I will spin up a container with the latest version and import the backup files.
Thx and best regards

**Reply by michael-genson** [reply]
Did you try the steps outlined above?

https://github.com/mealie-recipes/mealie/discussions/3081#discussioncomment-8306530
